rust/src/pgsql/pgsql.rs

```diff
@@ -203,8 +216,9 @@ impl PgsqlState {
             let mut index = self.tx_index_completed;
             for tx_old in &mut self.transactions.range_mut(self.tx_index_completed..) {
                 index += 1;
-                if tx_old.tx_state < PgsqlTransactionState::ResponseDone {
-                    tx_old.tx_state = PgsqlTransactionState::FlushedOut;
+                if tx_old.tx_req_state < PgsqlTxReqProgress::TxReqDone && tx_old.tx_res_state < PgsqlTxResProgress::TxResDone {
```

Comment:
how common will this be? Will we not more commonly have the `Req` state as `TxReqDone`? AFAICS in most cases when we create a tx the req state will already be set to done?